crates/ruff_cli/src/lib.rs

```diff
@@ -213,13 +228,7 @@ fn check(args: CheckArgs, log_level: LogLevel) -> Result<ExitStatus> {
         loop {
             match rx.recv() {
                 Ok(event) => {
-                    let paths = event?.paths;
-                    let py_changed = paths.iter().any(|path| {
-                        path.extension()
-                            .map(|ext| ext == "py" || ext == "pyi")
-                            .unwrap_or_default()
-                    });
-                    if py_changed {
+                    if has_file_changed(&event?.paths) {
```

Comment:
I think we have to recompute most of the variables defined above starting from line 115 whenever a `pyproject.toml` changes. For example, you can configure the output format in the settings and we should respect that change if we support reloading configuration files.

Reply:
Absolutely! Is there now something more to be recalculated other than pyproject_strategy?

Reply:
I'm not too familiar with that code but it is probably the `pyproject_strategy` and all data that gets derived from it (all settings that are supported in the configuration and aren't CLI options only).